cellular_configs/v1alpha1/amfconfig_types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AmfConfigStatus defines the observed state of AmfConfig
+type AmfConfigStatus struct {
+}
+
+// +kubebuilder:object:root=true
+type AmfConfig struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   AmfConfigSpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status AmfConfigStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// AmfConfigList contains a list of amfConfigs
+type AmfConfigList struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Items           []AmfConfig `json:"items" yaml:"items"`
+}
+
+// AmfConfigSpec defines the characteristics of a deployment of a network function
+type AmfConfigSpec struct {
+	// amfName defines name of the amf NF
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	AmfName string `json:"amfName" yaml:"amfName"`
+	// ServedGuami defines information related to GUAMI
+	ServedGuami []ServedGuamiConfig `json:"servedGuami" yaml:"servedGuami"`
+}
+
+// ServedGuami information related to GUAMI
+type ServedGuamiConfig struct {
+	// PlmnId defines the PLMN Identifier
+	PlmnId PlmnId `json:"plmnId" yaml:"plmnId"`
+	// AmfId defines the AMF Identifier
+	AmfId AmfId `json:"amfId" yaml:"amfId"`
+}
+
+// PlmnId defines the Public Land Mobile Network Identifier
+type PlmnId struct {
```

Comment:
why can we not reuse this from common config  or vice versa?
This is duplication.

Reply:
we can use json:inline for this to inherit the json tags.

Reply:
PLMN is also needed for free5gc SMF --- so it seems to be something that shouldn't be specific to AMF

Reply:
@henderiw This is not really duplication, how should I mention the `amdId` under the corresponding `plmnId` if I take the `plmnId` from the common crd? Also here it is GUAMI --> Globally Unique AMF ID

@s3wong in SMF we need only plmnID

```
plmnList:
  - mcc: "208"
    mnc: "93"
```

Reply:
if you use json:inline you don't need to redefine these parameters.